Eventing updates for Release/7.1 (#2696)
couchbase/docs-server: modules/eventing/pages/eventing-api.adoc

NOTE: Changes to the Eventing Function definition files made outside of this REST API or the interactive UI are only supported for version 7.0 (and above) if the Eventing schemas located in https://github.com/couchbase/eventing/tree/master/parser are adhered to.

As of version 7.1 Eventing now supports full xref:eventing-rbac.adoc[Eventing Role-Based Access Control] (RBAC). 

If the sample REST API shows two forms:

> * The first format is for privileged Eventing functions with a "Function Scope" of *+*+.+*+*, these functions can only be made or managed by users with the "Full Admin", "Eventing Full Admin" role.  In addition this is the default "Function Scope" for all functions after an upgrade from a prior version.

Reviewer: ```suggestion
* The first format is for privileged Eventing functions with a "Function Scope" of *+*+.+*+*; these functions can only be made or managed by users with the "Full Admin", "Eventing Full Admin" role.  
In addition this is the default "Function Scope" for all functions after an upgrade from a prior version.
```

> * The second format uses RBAC and you must pass the parameters *bucket=[fs_bucket]&scope=[fs_scope]* to specify the *bucket* and the *scope* of the "Function Scope" to the REST call.  In addition due to the use or *&* and *?* characters you will need to quote your REST call on the command line.

Reviewer: ```suggestion
* The second format uses RBAC and you must pass the parameters *bucket=[fs_bucket]&scope=[fs_scope]* to specify the *bucket* and the *scope* of the "Function Scope" to the REST call.  In addition, due to the use or *&* and *?* characters, you will need to quote your REST call on the command line.
```

> If the sample REST API shows just a single format the REST API call do not require the parameters *bucket=[fs_bucket]&scope=[fs_scope]* and the action is fully determined by the *username* and *password* credentials passed to the REST call.

Reviewer: ```suggestion
If the sample REST API shows just a single format, then the REST API call does not require the parameters *bucket=[fs_bucket]&scope=[fs_scope]* and the action is fully determined by the *username* and *password* credentials passed to the REST call.
```
